core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -91,6 +97,64 @@ public DeterministicKey(ImmutableList<ChildNumber> childNumberPath,
         this.keyCrypter = checkNotNull(crypter);
     }
 
+    /**
+     * Return the fingerprint of this key's parent as an int value, or zero if this key is the
+     * root node of the key hierarchy.  Raise an exception if the arguments are inconsistent.
+     * This method exists to avoid code repetition in the constructors.
+     */
+    private int ascertainParentFingerprint(DeterministicKey parentKey, byte[] parentFingerprint)
```

Comment:
Int seems better than byte array to me, but I could be missing a reason otherwise.  This file seems to predate github hosting this repository, so the original author is missing.  Personally I'm leaning toward all fingerprints as ints everywhere, including the return values of `getFingerprint()` and `getParentFingerprint()`, but obviously that means changing the existing interface.  If we're going to change it for the better, we should get it over with.

You're the decider, Mike Hearn.  Say how you want it and I'll put it that way.